Makefile

```diff
@@ -0,0 +1,239 @@
+
+# Install shellcheck
+$(SHELLCHECK_BIN):
+	@echo "installing shellcheck $(SHELLCHECK_VERSION)"
+	@mkdir -p "$(SHELLCHECK_PATH)"
+	@curl -sSfL "https://github.com/koalaman/shellcheck/releases/download/$(SHELLCHECK_VERSION)/$(SHELLCHECK_ARCHIVE)" \
+		| tar -x -J --strip-components=1 -C "$(SHELLCHECK_PATH)" "shellcheck-$(SHELLCHECK_VERSION)/shellcheck"
+
+# Install golangci-lint
+$(GOLANGCI_LINT_BIN):
+	@echo "installing golangci-lint v$(GOLANGCI_LINT_VERSION)"
+	@mkdir -p "$(GOLANGCI_LINT_PATH)"
+	@curl -sSfL "https://github.com/golangci/golangci-lint/releases/download/v$(GOLANGCI_LINT_VERSION)/$(GOLANGCI_LINT_ARCHIVE)" \
+		| tar -x -z --strip-components=1 -C "$(GOLANGCI_LINT_PATH)" "$(GOLANGCI_LINT_ARCHIVEBASE)/golangci-lint"
+
+.PHONY: lint
+# Run lint against the previous commit for PR and branch build
+# In dev setup look at all changes on top of master
+lint: $(GOLANGCI_LINT_BIN)
+	@echo "running golangci-lint"
+	@"${GOLANGCI_LINT_BIN}" run
+
+# Check scripts
+.PHONY: check_scripts
+ALLSCRIPTS := $(shell find . -not \( -path ./tools -prune \) -not \( -path ./build -prune \) -name '*.sh')
+check_scripts: $(SHELLCHECK_BIN)
+	@echo "running shellcheck"
+	@"$(SHELLCHECK_BIN)" ${ALLSCRIPTS}
+
+.PHONY: license-check
+# This is a bit convoluted but using a recursive grep on linux fails to write anything when run
+# from the Makefile. That caused the pull-request license check run from the github action to
+# always pass. The syntax for find is slightly different too but that at least works in a similar
+# way on both Mac and Linux. Excluding all .git* files from the checks.
+license-check:
+	@echo "checking license headers:"
+ifeq (darwin,$(OS))
+	$(shell mkdir -p build && find -E . -not \( -path './.git*' -prune \) -not \( -path ./build -prune \) -not \( -path ./tools -prune \) -regex ".*\.(go|sh|md|yaml|yml|mod)" -exec grep -L "Licensed to the Apache Software Foundation" {} \; > build/license-check.txt)
+else
+	$(shell mkdir -p build && find . -not \( -path './.git*' -prune \) -not \( -path ./build -prune \) -not \( -path ./tools -prune \) -regex ".*\.\(go\|sh\|md\|yaml\|yml\|mod\)" -exec grep -L "Licensed to the Apache Software Foundation" {} \; > build/license-check.txt)
+endif
+	@if [ -s "build/license-check.txt" ]; then \
+		echo "following files are missing license header:" ; \
+		cat build/license-check.txt ; \
+		exit 1; \
+	fi
+	@echo "  all OK"
+
+# Check that we use pseudo versions in master
+.PHONY: pseudo
+BRANCH := $(shell git branch --show-current)
+SI_REF := $(shell "$(GO)" list -m -f '{{ .Version }}' github.com/apache/yunikorn-scheduler-interface)
+SI_MATCH := $(shell expr "${SI_REF}" : "v0.0.0-")
+pseudo:
+	@echo "pseudo version check"
+	@if [ "${BRANCH}" = "master" ]; then \
+		if [ ${SI_MATCH} -ne 7 ]; then \
+			echo "YuniKorn references MUST all be pseudo versions:" ; \
+			echo " SI ref: ${SI_REF}" ; \
+			exit 1; \
+		fi \
+	fi
+	@echo "  all OK"
+
+# Build the example binaries for dev and test
```

Comment:
From here down, to the `clean` target, the targets should be removed and replaced with a target to call the make in the perf-tools.
When the soak test code is done we can add that as a new target in this file also.